packages/grafana-ui/src/components/uPlot/Plot.tsx

```diff
@@ -1,94 +1,107 @@
-import React, { useEffect, useLayoutEffect, useMemo, useRef } from 'react';
-import uPlot, { AlignedData, Options } from 'uplot';
+import React, { createRef } from 'react';
+import uPlot, { Options } from 'uplot';
 import { PlotContext } from './context';
-import { DEFAULT_PLOT_CONFIG, pluginLog } from './utils';
+import { DEFAULT_PLOT_CONFIG } from './utils';
 import { PlotProps } from './types';
-import usePrevious from 'react-use/lib/usePrevious';
+
+function sameDims(prevProps: PlotProps, nextProps: PlotProps) {
+  return nextProps.width === prevProps.width && nextProps.height === prevProps.height;
+}
+
+function sameData(prevProps: PlotProps, nextProps: PlotProps) {
+  return nextProps.data === prevProps.data;
+}
+
+function sameConfig(prevProps: PlotProps, nextProps: PlotProps) {
+  return nextProps.config === prevProps.config;
+}
+
+type UPlotChartState = { ctx: { plot: uPlot | null } };
```

Comment:
can we have this as a flat object? Can't see any value in having it nested

Reply:
and just pass the entire state down as the context value?

Reply:
the context will containt two values (plot, getPlotBoundingBox) doing it this way won't hurt:) 


```ts
const {plot, plotBBox} = this.state;
<...Provider vallue={{plot, plotBBox}} ...>
```

Honestly - Plot shouldn't have any state in the future - functionality wise it should be pretty much a dummy component.

Reply:
left as-is for now, since context value is an sCU boundary by referential integrity: https://reactjs.org/docs/context.html#caveats